### PR TITLE
feat(memory-v2): export listSkillEntries from skill-store

### PR DESCRIPTION
**Pull request by siddseethepalli**
## Summary
- Export `listSkillEntries(): SkillEntry[]` from `skill-store.ts` returning a sorted snapshot of cached skill entries.
- Add tests covering empty, populated, and mutation-safety paths.

Part of plan: memory-v2-sonnet-router.md (PR 4 of 10)

---

<a href="https://app.devin.ai/review/vellum-ai/vellum-assistant/pull/30169" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Comment by siddseethepalli** [comment]
Addressed in https://github.com/vellum-ai/vellum-assistant/pull/30550

File: assistant/src/memory/v2/skill-store.ts [hunk 1]
```diff
@@ -273,6 +273,22 @@ export function isSkillSlug(slug: string): boolean {
...
+ * for determinism. Returns a freshly allocated array on each call so callers
+ * cannot mutate the underlying cache.
+ *
+ * The cache is replaced atomically by `seedV2SkillEntries`, so a snapshot
+ * may be stale once a subsequent seed run completes. Callers that need
+ * up-to-the-moment state must re-call this after awaiting the seed.
+ */
+export function listSkillEntries(): SkillEntry[] {
+  if (!entries) return [];
+  return [...entries.values()].sort((a, b) =>
```
[the remark this answers is not in the document]
Comment (siddseethepalli) [reply assistant/src/memory/v2/skill-store.ts:287]: Addressed in https://github.com/vellum-ai/vellum-assistant/pull/30550
